components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -1925,6 +1928,137 @@ void JsonRpcService::GetERC1155TokenBalance(
       network_url, std::move(internal_callback));
 }
 
+void JsonRpcService::DiscoverAssets(
+    const std::string& chain_id,
+    const std::vector<std::string>&
+        account_addresses,  // TODO(nvonpentz) - Use AccountInfoPtr instead
+    std::vector<mojom::BlockchainTokenPtr> user_assets,
+    DiscoverAssetsCallback callback) {
+  if (chain_id != mojom::kMainnetChainId) {
+    std::move(callback).Run(
+        std::vector<mojom::BlockchainTokenPtr>(),
+        mojom::ProviderError::kMethodNotSupported,
+        l10n_util::GetStringUTF8(IDS_WALLET_METHOD_NOT_SUPPORTED_ERROR));
+    return;
+  }
+
+  if (account_addresses.empty()) {
+    std::move(callback).Run(
+        std::vector<mojom::BlockchainTokenPtr>(),
+        mojom::ProviderError::kInvalidParams,
+        l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+    return;
+  }
+
+  for (const auto& account_address : account_addresses) {
+    if (!EthAddress::IsValidAddress(account_address)) {
+      std::move(callback).Run(
+          std::vector<mojom::BlockchainTokenPtr>(),
+          mojom::ProviderError::kInvalidParams,
+          l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+      return;
+    }
+  }
+
+  auto internal_callback = base::BindOnce(
+      &JsonRpcService::OnGetAllTokensDiscoverAssets,
+      weak_ptr_factory_.GetWeakPtr(), chain_id, account_addresses,
+      std::move(user_assets), std::move(callback));
+
+  BlockchainRegistry::GetInstance()->GetAllTokens(
```

Comment:
if i'm not mistaken, this ultimately uses https://github.com/brave/brave-core/blob/f001b2d88f7ec308b49f5bb006045af13a2057cf/components/brave_wallet/browser/blockchain_list_parser.cc#L71 ? it looks like that list doesn't get passed through JSON sanitizer, but it probably should unless we are carefully auditing every update

also https://github.com/brave/brave-core/blob/f001b2d88f7ec308b49f5bb006045af13a2057cf/components/brave_wallet/browser/blockchain_list_parser.cc#L74 should be VLOG

Reply:
Updated in https://github.com/brave/brave-core/pull/13295/commits/55b6d2749a99d157ae04c987969349549c0098ca